### DISCUSSION TITLE
Using Allegro with pair_style/hybrid/overlay/kk and pair_style/zbl/kk

### DISCUSSION THREAD

**Discussion by mstapelberg** [mir-group/pair_nequip_allegro#46] [General]
Hi there, I was wondering if anyone has combined allegro potentials with another pair_style like a ZBL (for short range repulsion)? Or if there's a relatively straightforward way to set a minimum cutoff radius for allegro? 

I.e use ZBL for <1 Angstrom, allegro potential 1 for 1<X<2.5 Angstrom, and allegro potential 2 for X>2.5 Angstrom? Similar to the approach done here https://doi.org/10.1016/j.jnucmat.2018.06.005 , but replacing the EAM with Allegro. 

Thanks!
Myles

**Comment by Linux-cpp-lisp** [comment]
Hi @mstapelberg ,

Thanks for your interest in our code!

In general `pair_allegro` is compatible with `hybrid/overlay` with other pair styles, although of course it has to make sense that you would actually want to simulate with the potential that is the sum of your Allegro model and whatever other pair style you use.  Maybe if LAMMPS supports it directly you could do this kind of extremely granular cutoff setup, but we have no particular support for it at present.

ZBL and short range repulsion in particular, though, we support natively in `nequip` and Allegro.  You can see how to enable it here:
https://github.com/mir-group/nequip/blob/main/configs/minimal_pair.yaml

This also conveniently includes the ZBL term in the training process, simplifying the setup to get a meaningful summed potential.

**Reply by mstapelberg** [reply]
Hi @Linux-cpp-lisp 

Thanks so much for your quick reply and help! 

Great to know that ZBL and short range repulsion are natively supported. This will help things out a ton! Are there any specific types of training data that you'd recommend we add to high temperature AI-MD simulations? Also is there a reference you'd recommend I check out to learn more about how you implement the ZBL? 

Thanks,
Myles
